### PR TITLE
download_media: Special casing for imgur

### PR DESCRIPTION
**Pull request by ekimekim**
Handle imgur links, including links to albums/galleries by downloading the source image
(or for albums, all images) instead of the directly linked url.

### CODE REVIEW

**Comment by ekimekim** [comment]
This code is not yet tested.

File: common/common/media.py [hunk 1]
```diff
@@ -245,3 +273,110 @@ def _request(url, max_size, content_types):
...
+			These map to actual media and are stored in the usual way.
+		imgur.com/a/ID
+		imgur.com/gallery/ID
+			These map to collections of media.
+			Under the original URL we store a json file that lists imgur.com/ID urls
+			of the contents of the collection. Those urls are then downloaded and stored
+			in the usual way.
+	"""
+	parsed = urllib.parse.urlparse(url)
+	if parsed.hostname not in ("imgur.com", "i.stack.imgur.com"):
```
Comment (dcollinsn) [inline common/common/media.py:293]: What about `i.imgur.com`? (Mentioned in the docstring but not here)

Comment (ekimekim) [reply common/common/media.py:293]: `i.imgur.com` requires no special handling (it's a direct link to the image) so we don't need to handle it here.

Comment (ekimekim) [reply common/common/media.py:293]: I can clarify that in the docstring, you're right it does say this function "handles" that case.

**Review by chrusher — APPROVED** [review]
Looks ok to me

**Pull request merged**